### PR TITLE
feat: revamp select and add async paginated select

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Revamps the Select component, to prevent weird behaviour in some cases.
- Adds pagination to AsyncSelect
- Divides Select component into a AsyncSelect and Select component

**Note**
I've not added it anywhere to the admin yet, as I wanted some feedback before I go in a replace every instance across the codebase. There is several Stories that you can check out to see it in action instead 😄

### CODE REVIEW

File: src/components/molecules/new-select/selects/async-select/index.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import { SelectStyle } from "../../constants"
+import { SelectProvider } from "../../context"
+import { AsyncSelectProps } from "../../types"
+import { AsyncPaginate } from "./async-paginate-base"
+import { AsyncPaginateCreatable } from "./async-paginate-createable"
+
+const AsyncSelect = ({
+  value,
+  isClearable,
+  isCreateAble,
```
Comment (fPolic) [inline src/components/molecules/new-select/selects/async-select/index.tsx:26]: nit: should we follow consistent naming and rename this to `isCreateable` (also for `isSearchAble`)

**Review by fPolic — APPROVED** [review]
Nice work @kasperkristensen! LGTM 👍

File: src/components/molecules/new-select/components/control.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  innerProps,
+  selectProps: { menuIsOpen },
+  innerRef,
+}: ControlProps<T, IsMulti, GroupType>) => {
+  return (
+    <div
+      {...innerProps}
+      ref={innerRef}
+      className={clsx(
+        "bg-grey-5 inter-base-regular w-full p-3 cursor-text border flex border-grey-20 focus-within:shadow-input focus-within:border-violet-60 rounded-rounded",
```
Comment (pKorsholm) [inline src/components/molecules/new-select/components/control.tsx:20]: Have you talked to Ludvig about the border radius? I can see he has changed it, in the revamp in January it was 4px, however, in the price list, customer group design it's 8px for this and all other inputs. If we change it we should probably create an issue to align the remaining input fields to ensure the border radius is consistent.

```suggestion
        "bg-grey-5 inter-base-regular w-full p-3 cursor-text border flex border-grey-20 focus-within:shadow-input focus-within:border-violet-60 rounded",
```

**Review by pKorsholm — APPROVED** [review]
As always inspiring work Kasper 😍 LGTM

File: src/components/molecules/new-select/types/index.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  onChange: (newValue: unknown, actionMeta: ActionMeta<unknown>) => void
+  menuPortalTarget?: HTMLElement | null
+  placeholder?: string
+  searchPlaceholder?: string
+  label: string
+  required?: boolean
+  tooltip?: string | React.ReactElement
+}
+
+type MultiProps =
```
Comment (adrien2p) [inline src/components/molecules/new-select/types/index.ts:24]: this should probably be along the line of to enforce the type:

```ts
type MultiProps <TMulti extends boolean> = (TMulti extends true ? {
  isMulti: true
  hasSelectAll?: boolean
} : {
  isMulti: false
}) & BaseProps
```

File: src/components/molecules/new-select/types/index.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      hasSelectAll?: boolean
+    } & BaseProps)
+  | ({
+      isMulti?: false
+      hasSelectAll?: never
+    } & BaseProps)
+
+// If the Select is createable then the onCreate prop is required,
+// if it's not createable then the onCreate prop should be undefined
+type CreateAbleProps =
```
Comment (adrien2p) [inline src/components/molecules/new-select/types/index.ts:36]: same comment as the previous one here

**Comment by adrien2p** [comment]
Beautiful work @kasperkristensen :)

**Comment by kasperkristensen** [comment]
Closing as this component has been revamped in a later PR

**Pull request closed without merging**